src/lib/mongoBackend/MongoCommonSubscription.cpp

```diff
@@ -202,33 +202,43 @@ void setEntities(const Subscription& sub, BSONObjBuilder* b)
   BSONArrayBuilder entities;
   for (unsigned int ix = 0; ix < sub.subject.entities.size(); ++ix)
   {
-    EntID en = sub.subject.entities[ix];
+    EntID       en        = sub.subject.entities[ix];
+    std::string finalId;
+    std::string finalType;
+    std::string isIdPattern;
+    bool        isTypePattern = false;
```

Comment:
@crbrox will tell, but I guess that this is not easy to change... Remember that due to regacy reasons the id pattern is a string in many internal places.

What could probably be done at least is to add a comment explainin this "justified weridness".